src/server/visual-diff-reporter.js

```diff
@@ -0,0 +1,74 @@
+import { basename, dirname, join } from 'path';
+import { readdirSync, rmSync } from 'node:fs';
+
+function getDiffPaths(dir, _diffPaths) {
+	_diffPaths = _diffPaths || [];
+	const paths = readdirSync(dir, { withFileTypes: true });
+	paths.forEach(path => {
+		const full = join(dir, path.name);
+		const base = basename(path.name);
+
+		if (path.isDirectory()) {
+			if (base === 'pass' || base === 'fail') _diffPaths.push(full);
+			getDiffPaths(full, _diffPaths);
+		} else {
+			if (base === 'report.json') _diffPaths.push(full);
+		}
+	});
+	return _diffPaths;
+}
+
+function cleanupFiles(rootDir, testFiles, updateGoldens, runSubset) {
+	const vdiffDir = join(rootDir, '.vdiff');
+	let pathsToClear = [];
+
```

Comment:
The logic below is:
- with `--filter` and  `--golden`
  - Delete all directories at the location of the test files to be run (will delete too much if there are sibling and nested test files)
- with `--filter`
  - Delete all `pass` and `fail` directories and `reports` within the location of the test files to be run (will delete too much if there are sibling and nested test files)
- with `--golden`
  - Delete `.vdiff`
- with nothing
  - Delete all `pass` and `fail` directories and `reports` within `.vdiff`